### PR TITLE
fix KV issue

### PR DESCRIPTION
**Pull request by ashamis**
==> Running tests

### CODE REVIEW

File: src/kv/kv.h [hunk 1]
```diff
@@ -1035,9 +1033,11 @@ namespace kv
               // because of this we are returning NoVersion
               h->add_result(req_id, NoVersion);
             }
+            committed = true;
```
Comment (eddyashton) [inline src/kv/kv.h]: I think we should just set this once, at line 1018. And we should also move the similar line in `commit_reserved`, to be after we've retrieved a commit version.

**Comment by achamayou** [comment]
link #834

File: src/kv/kv.h [hunk 2]
```diff
@@ -998,8 +998,6 @@ namespace kv
       if (committed)
         throw std::logic_error("Transaction already committed");
 
-      committed = true;
-
       if (view_list.empty())
       {
         success = true;
```
Comment (eddyashton) [inline src/kv/kv.h]: Ah I missed this before - we also need to set `committed = true;` here, if we succeed due to an empty view_list.

**Comment by eddyashton** [comment]
After further discussion, I believe we've agreed that this is superceded by #866, which should avoid the exception _and_ the infinite loop, and should be safe (though fragile) in the PBFT case.

**Comment by ashamis** [comment]
this is superseded by #866

**Pull request closed without merging**